### PR TITLE
Simple raft implementation (raft-lite)

### PR DESCRIPTION
**Pull request by oestrich**
- Simple leader election
- Add in nodes
- Handle leader node dying

Todo:
- [ ] Link raft processes to know when the raft process dies? Start re-election

### CODE REVIEW

File: lib/raft/server.ex [hunk 1]
```diff
@@ -169,6 +194,10 @@ defmodule Raft.Server do
     end
   end
 
+  @doc """
+  Check if the ndoe has voted in this term
```
Comment (mattmenefee) [inline lib/raft/server.ex]: `node` instead of `ndoe`?

**Comment by tielur** [comment]
Is this implementation based off of the [Raft](https://github.com/toniqsystems/raft) library?

**Comment by oestrich** [comment]
It's based off of me reading the paper. I only want/need leader election and not the distributed log, plus its fun to write out. I reviewed hex and that library looked too new to use so I went with this.

**Comment by oestrich** [comment]
[Raft protocol - leader election](https://trello.com/c/ALQQwi8K/141-raft-protocol-leader-election)

**Pull request merged**